### PR TITLE
Makes the UWP VideoView more MVVM friendly

### PR DESCRIPTION
**Pull request by kakone**
### Description of Change ###
I replaced the SwapChainOptions and MediaPlayer properties to dependency properties. So It's easier to code in MVVM way with the UWP video view (I updated the sample to show the changes).
There is no breaking changes, property names stays the same.

Other change : the VideoView class inherits now from the Control class instead of UserControl. The developper can now updates the ControlTemplate if he wants. The default control template in Generic.xaml contains only the SwapChainPanel.

### Platforms Affected ### 
- UWP

### CODE REVIEW

File: LibVLCSharp/LibVLCSharp.csproj [hunk 1]
```diff
@@ -48,6 +48,7 @@ LibVLC needs to be installed separately, see VideoLAN.LibVLC.* packages.</Descri
     <PackageIcon>icon.png</PackageIcon>
     <PackageReleaseNotes>https://code.videolan.org/videolan/LibVLCSharp/blob/master/NEWS</PackageReleaseNotes>
     <PackageTags>libvlc;vlc;videolan;native;c/c++;video;audio;player;media;mediaplayer;codec;ffmpeg;xamarin;graphics;ios;android;linux;windows;macos;cross-platform</PackageTags>
+    <GenerateLibraryLayout>true</GenerateLibraryLayout>
```
Comment (mfkl) [inline LibVLCSharp/LibVLCSharp.csproj]: It used to work without this. Why is this needed now?

File: Samples/LibVLCSharp.UWP.Sample/MainPage.xaml [hunk 2]
```diff
@@ -8,8 +8,11 @@
     xmlns:lvs="using:LibVLCSharp.Platforms.UWP"
     mc:Ignorable="d"
     Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+    <Page.DataContext>
+        <local:MainViewModel />
+    </Page.DataContext>
 
     <Grid>
-      <lvs:VideoView x:Name="VideoView"></lvs:VideoView>
+      <lvs:VideoView SwapChainOptions="{Binding SwapChainOptions, Mode=TwoWay}" MediaPlayer="{Binding MediaPlayer}"></lvs:VideoView>
```
Comment (mfkl) [inline Samples/LibVLCSharp.UWP.Sample/MainPage.xaml]: Why `TwoWay` binding mode?

**Comment by mfkl** [comment]
I get
```
Error: Cannot get 'SwapChainOptions' value (type 'Object') from type 'LibVLCSharp.UWP.Sample.MainViewModel'. BindingExpression: Path='SwapChainOptions' DataItem='LibVLCSharp.UWP.Sample.MainViewModel'; target element is 'LibVLCSharp.Platforms.UWP.VideoView' (Name='null'); target property is 'SwapChainOptions' (type 'IReferenceArray`1<String>').
```

When running the UWP sample from your branch.

**Comment by kakone** [comment]
> I get
> 
> ```
> Error: Cannot get 'SwapChainOptions' value (type 'Object') from type 'LibVLCSharp.UWP.Sample.MainViewModel'. BindingExpression: Path='SwapChainOptions' DataItem='LibVLCSharp.UWP.Sample.MainViewModel'; target element is 'LibVLCSharp.Platforms.UWP.VideoView' (Name='null'); target property is 'SwapChainOptions' (type 'IReferenceArray`1<String>').
> ```
> 
> When running the UWP sample from your branch.

Not really an error because you can continue but I can easily fix it by adding a public getter in the ViewModel.

File: Samples/LibVLCSharp.UWP.Sample/MainPage.xaml (hunk 2 above)
Comment (kakone) [reply Samples/LibVLCSharp.UWP.Sample/MainPage.xaml]: I would like to use `OneWayToSource` binding mode to send the `SwapChainOptions` to the ViewModel, but `OneWayToSource` binding doesn't exist in UWP. So, I used `TwoWay` binding mode.

Perhaps it might be better to create an event with the SwapChainOptions in the event data, and I can keep a non-bindable SwapChainOptions property.

File: LibVLCSharp/LibVLCSharp.csproj (hunk 1 above)
Comment (kakone) [reply LibVLCSharp/LibVLCSharp.csproj]: With the add of the `Generic.xaml` file, you need this to generate the `Generic.xbf` file (XAML binary format) and place it in the good subfolder. Without this, you would get an exception. In UWP class libraries, this option is set by default.
For the other librairies (Android, iOS, ...), there is no incidence.

**Comment by mfkl** [comment]
Indeed, the video didn't load for some reason first time I tried, works fine now.

> I can easily fix it by adding a public getter in the ViewModel.

Yes please :)

This swapchainoptions ugly API will go away for libvlcsharp 4.0 as we will use d3d callbacks FYI.

File: LibVLCSharp/LibVLCSharp.csproj (hunk 1 above)
Comment (mfkl) [reply LibVLCSharp/LibVLCSharp.csproj]: Ok.

File: Samples/LibVLCSharp.UWP.Sample/MainPage.xaml (hunk 2 above)
Comment (mfkl) [reply Samples/LibVLCSharp.UWP.Sample/MainPage.xaml]: Yes, an event will probably be better (I usually try to avoid `TwoWay` binding mode).

**Comment by kakone** [comment]
Changes made, I added an Initialized event.

File: LibVLCSharp/LibVLCSharp.csproj [hunk 3]
```diff
@@ -56,6 +56,9 @@ LibVLC needs to be installed separately, see VideoLAN.LibVLC.* packages.</Descri
   <PropertyGroup Condition="$(NETSTANDARD_ONLY)=='true'">
     <TargetFrameworks>netstandard2.0;netstandard1.1</TargetFrameworks>
   </PropertyGroup>
+  <PropertyGroup Condition="$([MSBuild]::IsOsPlatform('Windows'))">
+    <GenerateLibraryLayout>true</GenerateLibraryLayout>
```
Comment (mfkl) [inline LibVLCSharp/LibVLCSharp.csproj]: I guess this is fine, but since it's not (supposed to be) doing anything with any TFM other than the UAP one, you could have use the TFM in the condition instead of `IsOsPlatform`. As you wish though, no strong feelings here.

Comment (kakone) [reply LibVLCSharp/LibVLCSharp.csproj]: You are right, I changed it (cf d9d745ab2d185dfad7279e1f19a935b4750a34d6).

**Comment by mfkl** [comment]
Squashed and merged https://github.com/videolan/libvlcsharp/commit/60e481bc6afc34c6c9f659561d8d43dbc89cd2bb Thank you!
